### PR TITLE
feat: studio ui

### PR DESCRIPTION
**Pull request by casserni**
Some things from feat/ui were merged and some weren't So I cherry picked the ones that haven't gotten in yet

### CODE REVIEW

File: src/Input.tsx [hunk 1]
```diff
@@ -43,9 +43,11 @@ const Input: React.FunctionComponent<IInput> = React.forwardRef<HTMLInputElement
 
   return (
     <Box
-      px={3}
-      py={2}
+      p="10px"
```
Comment (wmhilton) [inline src/Input.tsx]: WHOA. What happened to using the styled-system sizes or whatever they're called?

Comment (casserni) [reply src/Input.tsx]: The gaps are too large, we don't have a sizing that maps to 10px right now

Comment (casserni) [reply src/Input.tsx]: In general right now they aren't very practical until we remap them
1=> 4px
2=> 8px
3=> 16px
4=>32px

File: src/Toast/ToastContent.tsx [hunk 2]
```diff
@@ -37,39 +38,69 @@ const ToastContent = React.forwardRef<HTMLElement, IToastContentProps>(function
...
   const showCloseIcon = closeIcon !== false;
 
+  const toastActions = actions.map((action, index) => {
+    const { label, onClick = noop, ...buttonProps } = action;
+    return (
+      <Button
+        key={index}
+        m="5px"
+        color={theme.toastFg}
+        backgroundColor="rgba(138,155,168,0.25)"
```
Comment (P0lip) [inline src/Toast/ToastContent.tsx]: Shouldn't this be themeable?

File: src/Toast/ToastContent.tsx [hunk 3]
```diff
@@ -37,39 +38,69 @@ const ToastContent = React.forwardRef<HTMLElement, IToastContentProps>(function
...
+        border="transparent"
+        {...buttonProps}
+        css={{ ':hover': { backgroundColor: 'rgba(138,155,168,0.15)', ':active': { border: 'transparent' } } }}
```
Comment (P0lip) [inline src/Toast/ToastContent.tsx]: Ditto

File: src/Toast/ToastContent.tsx [hunk 4]
```diff
@@ -37,39 +38,69 @@ const ToastContent = React.forwardRef<HTMLElement, IToastContentProps>(function
...
+  const toastActions = actions.map((action, index) => {
+    const { label, onClick = noop, ...buttonProps } = action;
+    return (
+      <Button
+        key={index}
+        m="5px"
+        color={theme.toastFg}
+        backgroundColor="rgba(138,155,168,0.25)"
+        border="transparent"
+        {...buttonProps}
```
Comment (P0lip) [inline src/Toast/ToastContent.tsx]: Seems like we let `css` from `buttonProps` slip through

File: src/Toast/ToastContent.tsx [hunk 5]
```diff
@@ -78,14 +109,17 @@ ToastContent.displayName = 'ToastContent';
 /**
  * STYLE
  */
-export const toastContentStyles = (theme: ITheme['toast']): IBoxCSS => {
+export const toastContentStyles = (props: IToastContentProps, theme: ITheme['toast']): IBoxCSS => {
+  const type: any = props.type;
```
Comment (P0lip) [inline src/Toast/ToastContent.tsx]:
```suggestion
  const type: keyof ITheme['toast'] = props.type;
```
Don't remember whether toast exists on the object, if not, rename it.

Comment (casserni) [reply src/Toast/ToastContent.tsx]: added

File: src/Toast/ToastContent.tsx (hunk 4 above)
Comment (casserni) [reply src/Toast/ToastContent.tsx]: now merges if existing

File: src/Toast/ToastContent.tsx (hunk 2 above)
Comment (casserni) [reply src/Toast/ToastContent.tsx]: added

**Pull request merged**